### PR TITLE
Add relevant planned enhancements

### LINKED ISSUE #260: [PE-D][Tester C] Error message is too long for add customer command
Error message is too long (8 lines long)

-------------
Labels: `type.FeatureFlaw` `severity.Low`
original: keithxun/ped#1

### LINKED ISSUE #261: [PE-D][Tester D] The tags are irrelevant to the context
The tags such as colleagues, family, ... might not suitable in this context. Maybe can change the tag to more relevant one such as repeatCustomer, VIP and prioritisedShipping.

-------------
Labels: `severity.Low` `type.FeatureFlaw`
original: zyng233/ped#3

### LINKED ISSUE #271: [PE-D][Tester C] Incorrect error message for index in edit order command
- Expected: Invalid order index
- Actual: Invalid command

-------------
Labels: `type.FeatureFlaw` `severity.Medium`
original: keithxun/ped#6

### LINKED ISSUE #290: [PE-D][Tester C] Email validation for .domainName
- Email validation may want to include .com, .sg or .domainName

-------------
Labels: `severity.Low` `type.FunctionalityBug`
original: keithxun/ped#13

### PR DESCRIPTION
**Pull request by awdse22**
Close #290 
Close #271 
Close #261 
Close #260

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -1210,12 +1210,45 @@ testers are expected to do more *exploratory* testing.
...
+      Example: deletecustomer 1
+      ```
+    - **Solution:** Shorten the error message to state that the index should be a positive integer.
+    - **Rationale:** Stating that the command format is invalid in the first line of the error message may confuse the user in a case where they type an invalid index such as `1a` by mistake, as it is only the parameters which are incorrect in this case. Making the error message more concise also provides users with a better experience.
+
+
+4. **Improve display of longer messages or entries**
+   - **Problem:** Currently, longer messages or entries can be cut off, especially for users with smaller resolution/larger scale.
+   - **Example:** ![longMessages](images/longMessages.png)
+   - **Solution:** Improve text wrapping and allow customisable/resizeable GUI.
```
Comment (FooChao) [inline docs/DeveloperGuide.md:1237]: resizeable ui as in resize individual ui parts like the one u reviewing? for now it is quite unclear becasue the first thing i think of is that we can resize the whole thing now already

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -1210,12 +1210,45 @@ testers are expected to do more *exploratory* testing.
...
+   - **Rationale:** Longer messages/entries being cut off may be inconvenient for certain users (e.g. being required to scroll left/right to read an error message). Hence, providing text wrapping, allowing users to customise the GUI (e.g. whether they prefer to display an ellipsis, wrap text, etc.) and resize the GUI will give them more freedom to adjust based on their preferences and for convenience.
+
+
+5. **Include a warning if there is only one domain label provided for email**
+   - **Problem:** Currently, the `EMAIL` parameter accepts emails with only one domain label e.g. `johndoe@gmail`
+   - **Solution:** Allow emails with a single domain label, but provide a warning for it.
+   - **Rationale:** It is possible for some email address to have only one domain label, but we deem it unlikely to happen for the target users of this application. It is more likely that the user may forget to include a second domain label (e.g. missing `.com` in `johndoe@gmail.com`).
```
Comment (FooChao) [inline docs/DeveloperGuide.md:1244]:
```suggestion
   - **Rationale:** It is possible for some email address to have only one domain label, but we deem it unlikely to happen for the target users of this application. It is more likely that the user may have forgotten to include a second domain label (e.g. missing `.com` in `johndoe@gmail.com`).
```

Comment (FooChao) [reply docs/DeveloperGuide.md:1244]: not too sure about this one

**Review by FooChao — COMMENTED** [review]
Some small nitpicks

**Review by kaixuan477 — APPROVED** [review]

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (awdse22) [reply docs/DeveloperGuide.md:1237]: yes i meant resizing individual ui parts. Will update to make it clearer

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (awdse22) [reply docs/DeveloperGuide.md:1244]: "may have forgotten" is more or talking about something that has happened, so i don't think it's really correct

Comment (FooChao) [reply docs/DeveloperGuide.md:1244]: but aren't we saying that when they put one domain label referring to previous sentence so it kind of happened

Comment (FooChao) [reply docs/DeveloperGuide.md:1244]: but small issue if u think it is ok just merge

**Pull request merged**